### ISSUE TITLE
Issues with Dockerfile Conversion from Debian-based to CentOS 7

### ISSUE THREAD

**Issue by luoqh17** [bunop/isoSegmenter#7]
I'm currently working on a project involving Docker and CentOS 7. Specifically, I am in the process of converting a Dockerfile that was originally intended for a Debian-based system to work with CentOS 7 instead.

This endeavor has entailed changing apt-get commands in the original Dockerfile to their corresponding yum commands, since CentOS 7 relies on the yum package management system. While doing this, however, I've run into a couple of significant issues.
Firstly, it appears that certain packages (namely "libgd-dev" and "libgif-dev" from the Debian-based system) cannot be directly translated to their CentOS 7 counterparts. I've attempted to substitute "gd-devel" and "giflib-devel", respectively, but both seem to be unavailable in the CentOS 7 yum repository.

I've tried manually downloading and installing these libraries from the source to solve these issues, but I'd like to streamline this process and make it more manageable. Furthermore, I suspect that there may be other inconsistencies between the Debian and CentOS 7 package systems that I haven't yet encountered.

**Comment by bunop** [comment]
Dear @luoqh17 , 

I'm not sure if I have understood your problem: this project is already build in [Docker Hub](https://hub.docker.com/r/bunop/isosegmenter), so you could get and run this old image in a new system simply with:

```bash
docker pull bunop/isosegmenter:latest
docker run -ti --rm bunop/isosegmenter:latest bash
```

without worrying about how this image was build, and this is the main advantages of using docker (or singularity). If you need to create a `centos:7` image, ideally you can do like this:

```dockerfile
FROM centos:7

RUN yum update -y && \
    yum install -y giflib-devel gd-devel
```

However, this project can only run in old system since is written in Python 2 and the `gd` dependency was never implemented to work with Python 3.

Hope this helps

**Comment by luoqh17** [comment]
Thank you so much for your prompt response and this great project. I'm fairly new to Docker and I'm not yet fully familiar with its operations. Following your assistance, I performed the command 

` docker run -ti --rm bunop/isosegmenter:latest bash `

and I encountered this issue:
"docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "cgroup" to rootfs at "/sys/fs/cgroup": stat /docker/4f5afcf97888508235847269adb1c3617764a430ba5dacee14d6507779c42474: no such file or directory: unknown. ERRO[0000] error waiting for container: context canceled"

Could you possibly offer any advice on this? I would greatly appreciate it. Thank you very much!

**Comment by luoqh17** [comment]
As I proceeded with the `docker pull bunop/isosegmenter:latest`, the response was as follows:
"latest: Pulling from bunop/isosegmenter
Digest: sha256:60393351732365ff6a05a4bfb7cb7b827e03722376bba33c1536e706118e61a5
Status: Image is up to date for bunop/isosegmenter:latest
docker.io/bunop/isosegmenter:latest"
I thought this meant that I had successfully pulled down the container.

**Comment by bunop** [comment]
Some questions:

- What OS are you using? what distro?
- Which version of docker are you using?
- Are you executing docker as a non-privileged user? have you the right permission to run docker?
- Can you execute a simple command like `docker run hello-world`?

**Comment by luoqh17** [comment]
Thanks a lot for your reply! I am truly grateful for your help! 
OS information: `Linux version 6.2.15-300.fc38.x86_64  Red Hat 13.1.1-1, GNU ld version 2.39-9.fc38)`
Docker information: `Docker version 26.0.0, build 2ae903e` 
I am afraid that the problem could be due to my Docker setup on the server. I chose to execute the Docker commands with root privileges, and yet, encountered issues even when I ran `docker run hello-world`. Considering this, it is likely that my Docker configurations are causing me trouble.
I am going to look into resolving the configuration issues on my server first. Thanks again for your kind help.
